src/Application/Command/User/ChangeEmail/ChangeEmailCommand.php

```diff
@@ -24,4 +23,14 @@ public function __construct(string $userUuid, string $email)
         $this->userUuid = Uuid::fromString($userUuid);
         $this->email = Email::fromString($email);
     }
+
+    public function getUserUuid(): UuidInterface
```

Comment:
In PHP there's no option to mark a public attribute as `readonly` like  [in other languages](https://github.com/jorge07/billing-api/blob/master/src/application/useCase/transaction/create/command.ts#L5). This makes me so sad as makes mandatory the use of private properties and getters, and I'm really against getters in Value Objects. IMO it adds just noise.
PSALM allows you to define [readonly attributes](https://psalm.dev/docs/annotating_code/supported_annotations/#psalm-readonly-and-readonly) (I did not saw it in PHPStan documentation) something that I ended doing as convention with some teams in the past but we can have a proper check in the CI pipeline.
Thoughts?

Reply:
@cv65kr ping

Reply:
PHP had proposal for introducing readonly properties but is dead for some reason. https://wiki.php.net/rfc/readonly_properties

I was thinking about that, but introducing next library only for possibility to make properties `readonly` doesn't make sense for me. We can use magic methods to make properties readonly but for me is not elegant way. So getters are probably best option for us.

Maybe you have another plan with psalm? Then maybe that will be good point to start using that library.

Reply:
You can use [Psalm](https://psalm.dev/articles/immutability-and-beyond) with `@psalm-immutable` annotation

Reply:
I want avoid of psalm using. I am still thinking how resolve this problem.

Reply:
I was thinking about the getters. I'm not a big fan but I understand that others will want it, or needed, specially newer profiles. So I'm OK adding it but I've one petition: 
Can we at least, remove the `get` prefix?

Reply:
Convention name for getters with get prefix for me is good. 
First of all it is readable and understandable for developers which start adventure with this template. 
`$class->email()` say nothing to me. Methods should describe what they do.

I have read fews articles about using getters/setters and opinions are divided. Personally i think for this use case it is nothing wrong with getters. Naturally you must made decision, how this should look 😩

Reply:
I truly believe open source is about people so I'll not force my opinion when there's no agreement.

Let's bring @Lutacon to the conversation and see his opinion

Reply:
good approach

Reply:
In this case, if we have available read-only properties I think we all can agree we don't need the getters. I don't use in my apps getters and private properties on DTOs.

In my opinion, the `get` prefix doesn't add any more value to the method. I never use the prefix `get` in any of our applications (personal or work). We prefer to name them accordingly

Checking the post of @grachevko the option of using psalm looks nice. We can use psalm to have missing features that PHP doesn't have at this moment.